src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OpenTelemetry.Exporter.OpenTelemetryProtocol.csproj

```diff
@@ -30,6 +30,7 @@
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
     <PackageReference Include="Google.Protobuf" Version="$(GoogleProtobufPkgVer)" />
     <PackageReference Include="Grpc.Tools" Version="$(GrpcToolsPkgVer)" PrivateAssets="all" />
+    <PackageReference Include="OpenTelemetry.Extensions.PersistentStorage" Version="$(PersistentStoragePkgVer)" />
```

Comment:
This is the file-system based implementation, would a user ever want to switch to something else?

Here's a quick rundown of how we could make it extensible...

1) Depend on the abstractions package. Don't depend on any concrete implementation.
2) Do one or both of these things:
   * Look for PersistentBlobProvider in the IServiceProvider. If you find something, use it. If not, don't. If users want to turn on the feature they have to setup a provider in their services using a concrete implementation of their choosing
   * Provide some factory extensibility point on options or an extension `Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory` so users can supply one directly via code

Reply:
+1 to decouple this

Reply:
Went with this option @CodeBlanch suggested
> Look for PersistentBlobProvider in the IServiceProvider

Reply:
discussed this offline with @CodeBlanch. I think we should go with `an extension exposing a signature like Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory so users can supply one directly via code`. In cases when we do not want the same PersistentBlobProvider across all signals `Look for PersistentBlobProvider in the IServiceProvider` may cause issues.